ast/ddl.go

```diff
@@ -1048,6 +1050,106 @@ type TableOption struct {
 	UintValue uint64
 }
 
+func (n *TableOption) Restore(ctx *RestoreCtx) error {
+	switch n.Tp {
+	case TableOptionEngine:
+		ctx.WriteKeyWord("ENGINE ")
+		ctx.WritePlain("= ")
+		if n.StrValue != "" {
+			ctx.WritePlain(n.StrValue)
+		} else {
+			ctx.WritePlain("''")
+		}
+	case TableOptionCharset:
+		ctx.WriteKeyWord("DEFAULT ")
+		ctx.WriteKeyWord("CHARACTER ")
+		ctx.WriteKeyWord("SET ")
+		ctx.WritePlain("= ")
+		ctx.WriteKeyWord(n.StrValue)
+	case TableOptionCollate:
+		ctx.WriteKeyWord("DEFAULT ")
+		ctx.WriteKeyWord("COLLATE ")
+		ctx.WritePlain("= ")
+		ctx.WriteKeyWord(n.StrValue)
+	case TableOptionAutoIncrement:
+		ctx.WriteKeyWord("AUTO_INCREMENT ")
+		ctx.WritePlain("= ")
+		ctx.WritePlain(fmt.Sprint(n.UintValue))
```

Comment:
```suggestion
		ctx.WritePlainf("%d", n.UintValue)
```

(Similar for the rest, not going to comment on each one-by-one 🙃)